docs/user-guide/configure.md

```diff
@@ -254,6 +254,186 @@ But the following pattern would be reported as a warning:
+  }
+}
+```
+
+The following rules are configured via the `languageOptions` property:
+
+- [`at-rule-descriptor-no-unknown`](../../lib/rules/at-rule-descriptor-no-unknown/README.md)
+- [`at-rule-descriptor-value-no-unknown`](../../lib/rules/at-rule-descriptor-value-no-unknown/README.md)
+- [`at-rule-prelude-no-invalid`](../../lib/rules/at-rule-prelude-no-invalid/README.md)
+- [`declaration-property-value-no-unknown`](../../lib/rules/declaration-property-value-no-unknown/README.md)
+
+#### `AtRules`
+
+You can customize at-rules by defining their expected prelude and descriptors.
+For example, to support a new at-rule `@foo` with specific descriptors:
+
+```json
+{
+  "languageOptions": {
+    "syntax": {
+      "atRules": {
+        "foo": {
+          "prelude": "<custom-ident>",
+          "descriptors": {
+            "bar": "<number>",
+            "baz": "<color>"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+When `at-rule-descriptor-no-unknown` is enabled, the following patterns are considered problems:
+
+<!-- prettier-ignore -->
+```css
+@foo custom-ident { qux: 10; }
+```
+
+When `at-rule-descriptor-value-no-unknown` is enabled, the following patterns are considered problems:
+
+<!-- prettier-ignore -->
+```css
+@foo custom-ident { bar: red; }
+```
+
+In both cases, the following pattern is _not_ considered a problem:
+
+<!-- prettier-ignore -->
+```css
+@foo custom-ident { bar: 10; baz: red; }
+```
+
+#### `cssWideKeywords`
+
+You can add custom global keywords that are accepted as valid property values.
+For example, to add `-moz-initial` to the [wide keywords](https://drafts.csswg.org/css-values-4/#css-wide-keywords):
+
+```json
+{
+  "languageOptions": {
+    "syntax": {
+      "cssWideKeywords": ["-moz-initial"]
+    }
+  }
+}
+```
+
+When `declaration-property-value-no-unknown` is enabled, the following patterns are considered problems:
+
+<!-- prettier-ignore -->
+```css
+a { color: foo; }
+```
+
+The following patterns are _not_ considered problems:
+
+<!-- prettier-ignore -->
+```css
+a { color: -moz-initial; }
+```
+
+#### `properties`
+
+You can extend or modify the syntax for specific properties.
+For example, to support a custom `foo` property that accepts `<color>` values:
+
+```json
+{
+  "languageOptions": {
+    "syntax": {
+      "properties": { "foo": "<color>" }
+    }
+  }
+}
+```
+
+When `declaration-property-value-no-unknown` is enabled, The following patterns are considered problems:
+
+<!-- prettier-ignore -->
+```css
+a { foo: 10px; }
+```
+
+The following patterns are _not_ considered problems:
+
+<!-- prettier-ignore -->
+```css
+a { foo: red; }
+```
+
+#### `types`
+
+You can extend or modify the syntax for specific types.
+For example, to support a new function `--foo()` that accepts `<length-percentage>` values and use it within the top property:
+
+```json
+{
+  "languageOptions": {
+    "syntax": {
+      "types": { "--foo()": "--foo( <length-percentage> )" },
```

Comment:
question

Can you create new custom "syntax strings" using `types`?
e.g. `"types": { "<foo>": "<length-percentage>" },`

Reply:
Yes, for example:

```js
properties: { top: '| <bar>' },
types: { bar: '<color>' },
```

With this setup, `a { top: red; }` won’t trigger a warning.

Reply:
off-topic

So the follow-up question is should we accept those custom types in `syntax-string-no-invalid`?